sql/src/main/java/io/crate/protocols/postgres/BulkPortal.java

```diff
@@ -136,14 +134,13 @@ public void execute(ResultReceiver resultReceiver, int maxRows) {
         return executeBulk(portalContext.getExecutor(), plan, jobId, statsTables);
     }
 
-    private ListenableFuture<Void> executeBulk(Executor executor, Plan plan, final UUID jobId,
-                             final StatsTables statsTables) {
-        final SettableFuture<Void> future = SettableFuture.create();
-        List<? extends ListenableFuture<Long>> futures = executor.executeBulk(plan);
-        ListenableFuture<List<Long>> allAsList = Futures.allAsList(futures);
-        Futures.addCallback(allAsList, new FutureCallback<List<Long>>() {
-                @Override
-                public void onSuccess(@Nullable List<Long> result) {
+    private CompletableFuture<Void> executeBulk(Executor executor, Plan plan, final UUID jobId,
+                                                final StatsTables statsTables) {
+        final CompletableFuture<Void> future = new CompletableFuture<>();
+        List<CompletableFuture<Long>> futures = executor.executeBulk(plan);
+        CompletableFuture<List<Long>> allFuturesCollected = CompletableFutures.allSuccessfulOrFailedFuture(futures);
```

Comment:
Would it make sense to add a handler to each individual future here instead of using allSuccessfulOrFailedFuture? This is the only place where it is used (as far as I can see).